### PR TITLE
Exception handler

### LINKED ISSUE #3: Exception’s handling
Server should correctly process cases when client provides incorrect information. For example, client send http request GET http://localhost:8080/api/projects/111 where 111 is ID of project, but now such project doesn't exist in the database. In this case server should return 404 response code with message: "Project with id='111' not found". No error is appeared on server logs.

### PR DESCRIPTION
**Pull request by VadMack**

### CODE REVIEW

File: src/main/java/com/vadmack/mongodbtest/exception/handler/CustomExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.vadmack.mongodbtest.exception.handler;
+
+import com.vadmack.mongodbtest.exception.NotFoundException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+public class CustomExceptionHandler {
```
Comment (iv21dk) [inline src/main/java/com/vadmack/mongodbtest/exception/handler/CustomExceptionHandler.java:10]: This class is not required for now. You can use the following approach:
https://www.baeldung.com/exception-handling-for-rest-with-spring#3-responsestatusexceptionresolver

File: src/main/java/com/vadmack/mongodbtest/service/ProjectService.java [hunk 2]
```diff
@@ -31,7 +32,7 @@
 
     public ProjectDto findById(Long id) {
         return repository.findById(id).map(project -> modelMapper.map(project, ProjectDto.class))
-                .orElse(new ProjectDto());
+                .orElseThrow(() -> new NotFoundException(String.format("Project with id=%d not found", id)));
```
Comment (iv21dk) [inline src/main/java/com/vadmack/mongodbtest/service/ProjectService.java]: `orElseThrow(() -> new NotFoundException(String.format("Project with id=%d not found", id))` - this is a repeated code. I suggest to create a new method:
`Project getById(Long id)`
that would return an entity if it exists and thrown an exception if not.

File: src/main/java/com/vadmack/mongodbtest/service/ProjectService.java [hunk 3]
```diff
@@ -41,12 +42,16 @@ public void create(ProjectNoIdDto projectNoIdDto) {
...
         Project updatedProject = modelMapper.map(projectNoIdDto, Project.class);
         updatedProject.setId(id);
         repository.save(updatedProject);
     }
 
     public void delete(Long id) {
+        repository.findById(id)
+                .orElseThrow(() -> new NotFoundException(String.format("Project with id=%d not found", id)));
         Optional<Project> optionalProject = repository.findById(id);
```
Comment (iv21dk) [inline src/main/java/com/vadmack/mongodbtest/service/ProjectService.java]: You call `repository.findById(id)` twice. Please try to avoid

**Review by iv21dk — CHANGES_REQUESTED** [review]

File: src/main/java/com/vadmack/mongodbtest/exception/handler/CustomExceptionHandler.java (hunk 1 above)
Comment (VadMack) [reply src/main/java/com/vadmack/mongodbtest/exception/handler/CustomExceptionHandler.java:10]: "The same as the DefaultHandlerExceptionResolver, this resolver is limited in the way it deals with the body of the response — it does map the Status Code on the response, but the body is still null."

In the issue #3 it said "In this case server should return 404 response code with message: "Project with id='111' not found""

Comment (iv21dk) [reply src/main/java/com/vadmack/mongodbtest/exception/handler/CustomExceptionHandler.java:10]: You can resolve it using: 
server.error.include-message=always

**Review by iv21dk — APPROVED** [review]

**Pull request closed without merging**